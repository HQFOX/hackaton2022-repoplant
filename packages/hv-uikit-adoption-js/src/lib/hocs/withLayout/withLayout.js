import React from "react";
import Header from "components/layout/Header";
import Footer from "components/layout/Footer";

const getStyles = hasFooter => ({
  content: {
    padding: `${hasFooter ? 0 : "30px 15px"}` // grid padding
  }
});

const withLayout = (Component, hasFooter = false) => props => {
  const { content } = getStyles(hasFooter);

  return (
    <>
      <Header />
      <section id="content" style={{ ...content }}>
        <Component {...props} />
      </section>
      {hasFooter && <Footer />}
    </>
  );
};

export default withLayout;
