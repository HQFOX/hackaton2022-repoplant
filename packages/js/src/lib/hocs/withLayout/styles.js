import { createStyles } from "@material-ui/core";

const styles = () =>
  createStyles({
    section: {
      paddingTop: "50px",
      display: "flex"
    },
    component: {
      flexGrow: 1,
      padding: "30px 15px 0 15px"
    }
  });

export default styles;
