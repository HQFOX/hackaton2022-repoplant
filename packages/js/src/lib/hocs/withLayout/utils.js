export const findById = (data = [], id) => {
  for (let i = 0; i < data.length; i += 1) {
    const el = data[i];
    if (el.id === id) return el;

    const child = findById(el.data, id);
    if (child) return child;
  }
  return null;
};

export const findLeaf = item => (item.data ? findLeaf(item.data[0]) : item);

export const findFirstChildByParentId = (data = [], id) => {
  const parent = findById(data, id);
  return (parent && parent.data && findLeaf(parent.data[0]).id) || id;
};

export const navigationDepth = arr =>
  Array.isArray(arr)
    ? 1 + Math.max(...arr.map(el => navigationDepth(el.data)))
    : 0;
