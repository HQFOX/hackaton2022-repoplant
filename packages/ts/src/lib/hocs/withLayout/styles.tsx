import { makeStyles } from "@material-ui/core";

const useStyles = makeStyles(() => ({
  section: {
    minHeight: "100vh",
    paddingTop: "50px",
    display: "flex"
  },
  hasFooter: {
    minHeight: "calc(100vh - 40px)"
  },
  component: {
    flexGrow: 1
  },
  hasMargins: {
    padding: "30px 15px 0 15px"
  }
}));

export default useStyles;
