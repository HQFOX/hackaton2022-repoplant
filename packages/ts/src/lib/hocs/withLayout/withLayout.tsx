import React from "react";
import { Header, Footer } from "components/layout";

const getStyles = (hasFooter: boolean) => ({
  content: {
    padding: `${hasFooter ? 0 : "30px 15px"}`
  }
});

const withLayout = <P extends {}>(
  Component: React.ComponentType<P>,
  hasFooter = false
): React.FC<P> => props => {
  const { content } = getStyles(hasFooter);

  return (
    <>
      <Header />
      <section id="content" style={{ ...content }}>
        <Component {...(props as P)} />
      </section>
      {hasFooter && <Footer />}
    </>
  );
};

export default withLayout;
