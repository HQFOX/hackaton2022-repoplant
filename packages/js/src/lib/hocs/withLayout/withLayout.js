import React, { useState } from "react";
import { withStyles } from "@material-ui/core";
import Header from "components/layout/Header";
import Footer from "components/layout/Footer";
import VerticalNavigation from "components/layout/VerticalNavigation";
import NavigationContext from "../../NavigationContext";
import styles from "./styles";

const withLayout = (Component, hasFooter = false) =>
  withStyles(styles, { name: "withLayout" })(props => {
    const [isOpen, setIsOpen] = useState(false);
    const { classes } = props;

    const toggleOpen = () => {
      setIsOpen(prevState => !prevState);
    };

    return (
      <NavigationContext.Provider value={{ isOpen, toggleOpen }}>
        <Header />
        <div className={classes.section}>
          <VerticalNavigation />
          <main className={classes.component}>
            <Component {...props} />
          </main>
        </div>
        {hasFooter && <Footer />}
      </NavigationContext.Provider>
    );
  });

export default withLayout;
