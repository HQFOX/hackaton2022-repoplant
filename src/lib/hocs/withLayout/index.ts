import { useMediaQuery, useTheme } from "@material-ui/core";
import { isTopLevelPage } from "lib/utils/path";
import { useEffect, useState } from "react";
import { useLocation } from "react-router-dom";

export const useLayoutMargins = () => {
  const [margin, setMargin] = useState<number>(0);

  const theme = useTheme();
  const isMdUp = useMediaQuery(theme.breakpoints.up("md"));
  const { pathname } = useLocation();

  useEffect(() => {
    if (isMdUp && !isTopLevelPage(pathname)) {
      setMargin(84);
    } else {
      setMargin(44);
    }
  }, [theme, isMdUp, pathname]);

  return margin;
};

export { default } from "./withLayout";
