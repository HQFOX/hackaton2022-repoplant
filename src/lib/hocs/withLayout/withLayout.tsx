import React, { useContext } from "react";
import clsx from "clsx";
import { useLocation } from "react-router-dom";
import { HvContainer } from "@hv/uikit-react-core";
import { Footer, VerticalNavigation } from "components/layout";
import { AuthContext } from "lib/context/AuthContext";
import { isTopLevelPage } from "lib/utils/path";
import useStyles from "./styles";

const withLayout = <P extends {}>(
  Component: React.ComponentType<P>,
  hasFooter = false
): React.FC<P> => (props) => {
  const classes = useStyles();
  const { pathname } = useLocation();
  const { isAuthed } = useContext(AuthContext);
  const hasSecondLevel = !isTopLevelPage(pathname);

  return (
    <>
      <div
        className={clsx(classes.section, {
          [classes.hasFooter]: hasFooter,
        })}
      >
        <VerticalNavigation />
        <HvContainer
          maxWidth="xl"
          className={clsx(classes.container, {
            [classes.isAuthed]: isAuthed,
            [classes.hasSecondLevel]: isAuthed && hasSecondLevel,
          })}
        >
          <Component {...props} />
        </HvContainer>
      </div>
      {hasFooter && <Footer />}
    </>
  );
};

export default withLayout;
