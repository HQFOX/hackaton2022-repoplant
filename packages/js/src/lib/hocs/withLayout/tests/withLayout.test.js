import React from "react";
import { shallow } from "enzyme";
import withLayout from "../withLayout";

describe("withLayout", () => {
  let wrapper;

  const checkComponent = Component => {
    wrapper = shallow(<Component />);
    expect(wrapper).toMatchSnapshot();
  };

  it("is rendered correctly", () => {
    checkComponent(withLayout(React.Component));
    checkComponent(withLayout(React.Component, true));
  });
});
