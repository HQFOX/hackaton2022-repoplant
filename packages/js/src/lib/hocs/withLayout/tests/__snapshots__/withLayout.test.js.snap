// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withLayout is rendered correctly 1`] = `
<Component
  classes={
    Object {
      "component": "withLayout-component-2",
      "section": "withLayout-section-1",
    }
  }
/>
`;

exports[`withLayout is rendered correctly 2`] = `
<Component
  classes={
    Object {
      "component": "withLayout-component-4",
      "section": "withLayout-section-3",
    }
  }
/>
`;
